io/CMakeLists.txt

```diff
@@ -344,14 +344,27 @@ target_include_directories(${LIB_NAME} PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/inclu
 target_link_libraries("${LIB_NAME}" Boost::boost Boost::filesystem Boost::iostreams pcl_common pcl_io_ply)
 
 if(VTK_FOUND)
-  if(${VTK_VERSION} VERSION_LESS 9.0)
-    link_directories(${VTK_LINK_DIRECTORIES})
-    target_link_libraries("${LIB_NAME}" ${VTK_LIBRARIES})
-  else()
+  if(${VTK_VERSION} VERSION_GREATER_EQUAL 9.0)
     target_link_libraries("${LIB_NAME}" 
                           VTK::IOImage
                           VTK::IOGeometry
                           VTK::IOPLY)
+  else()
+    link_directories(${VTK_LINK_DIRECTORIES})
+    target_link_libraries("${LIB_NAME}"
+                          vtkIOImage
+                          vtkIOMPIImage
+                          vtkIOPLY
+                          vtkIOMPIParallel
+                          vtkRenderingFreeTypeFontConfig
+                          vtkRenderingMatplotlib
+                          vtkRenderingVolume
+                          vtkRenderingVolume${VTK_RENDERING_BACKEND}
```

Comment:
Do we need all of these libs? As it it the `pcl_io` lib, why e.g. `vtkRendering` is required?

Reply:
I did started with a "clean slate" and then added until it successfully linked. So my first answer would be yes, they are required. But I can try go over them once more.

Reply:
Would be nice if you can check, why these are required, maybe it is something which should doesn't belong to `pcl_io`